### PR TITLE
avoiding infinite loops

### PR DESCRIPTION
**Pull request by martinraison**
@leogrim I identified two places that could have caused infinite loops. I'm fixing it by updating the database before attempting to construct the emails. In the future we might want to persist these failures to the database (right now we airbrake).

### CODE REVIEW

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala [hunk 1]
```diff
@@ -208,6 +208,9 @@ class ElizaEmailCommander @Inject() (
   } else Future.successful()
 
   private def safeProcessEmail(messageThread: MessageThread, nonUserThread: NonUserThread, htmlBodyMaker: ProtoEmail => String, category: NotificationCategory): Future[Unit] = {
+    // Update records even if sending the email fails (avoid infinite failure loops)
+    // todo(martin) persist failures to database
+    db.readWrite{ implicit session => nonUserThreadRepo.setLastNotifiedAndIncCount(nonUserThread.id.get, clock.now()) }
```
Comment (leogrim) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala:213]: If you do this some users may silently never be notified, isn't it?

Comment (martinraison) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala:213]: correct - we'll have the airbrakes, though.

Comment (martinraison) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala:213]: I guess, if the email corresponds to a user adding an email to a conversation, we should think of a way to give him this feedback ("Sorry, your email could not be sent"). I wouldn't worry as much for digest emails, but persisting the failures may help us keep track of them (or maybe sending them to mixpanel?)

Comment (leogrim) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala:213]: we may want to retry at least once though, if Shoebox is out for some
unrelated reason, we'll have to recover manually

2014-05-19 17:59 GMT-07:00 martinraison notifications@github.com:

**Pull request merged**